shared/constants/chat.js

```diff
@@ -8,14 +8,17 @@ import type {NoErrorTypedAction} from './types/flux'
 
 export type MessageType = 'Text'
 export type FollowState = 'You' | 'Following' | 'Broken' | 'NotFollowing'
+export type MessageState = 'pending' | 'sent' | 'failed'
 
 export type Message = {
   type: 'Text',
   message: HiddenString,
   author: string,
   timestamp: number,
-  messageID: number,
+  messageID?: number,
   followState: FollowState,
+  messageState: MessageState,
+  outboxID?: ?string,
```

Comment:
:sheep: Is it true that a `Message` always has a `messageID` or an `outboxID`? I wonder if telling the types this (via a union type) would help catch future mistakes.